src/newcode/newCodeDefinitionService.ts

```diff
@@ -83,12 +83,21 @@ export class NewCodeDefinitionService {
     }
     const newCodeDefinition = this.getNewCodeDefinition(code2ProtocolConverter(workspaceFolder.uri));
     if (newCodeDefinition && this.cleanAsYouCode) {
+      const message = newCodeDefinition.newCodeDefinitionOrMessage;
       if (newCodeDefinition.isSupported) {
-        this.newCodeStatusBarItem.tooltip = `Showing issues on new code\n${newCodeDefinition.newCodeDefinitionOrMessage}`;
+        this.newCodeStatusBarItem.tooltip = `Showing issues on new code\n${message}`;
+      } else {
+        this.newCodeStatusBarItem.tooltip = `Showing all issues\n${message}`;
       }
       isSupportedForFile = newCodeDefinition.isSupported;
     } else {
-      this.newCodeStatusBarItem.tooltip = 'Showing all issues';
+      let reason = '';
+      if (!this.cleanAsYouCode) {
+        reason = '\nClean as You Code is disabled in settings';
+      } else if (!newCodeDefinition) {
+        reason = '\nThere is no new code definition for the project';
```

Comment:
I think the correct capitalization would be 'There is no New Code Definition for the project'.

Reply:
In SQ docs it's like this:
https://docs.sonarsource.com/sonarqube/10.1/project-administration/clean-as-you-code-settings/defining-new-code/#new-code-definition-options